client/cli/src/commands/build_spec_cmd.rs

```diff
@@ -75,14 +86,46 @@ impl BuildSpecCmd {
 		}
 		Ok(())
 	}
+
+	/// Sync the light client, export the sync state and run the command as per normal.
+	pub async fn run_export_sync_state<B, CL, BA>(
```

Comment:
I still don't like that this only works for light clients...

However, if you still want to follow this path. Make the function return an error when the user tries to run a full node when using this.

Reply:
Maybe a bad question but isnt the light client a subset of the full client? Can't we export the data with a full client, strip the data of unneeded content, and import it with a light client? So it could maybe work for light and full?

Reply:
You could build the data also on a full node. That was my point, why I don't like that it is only done on light clients :D